src/ipc/mem/PageStack.h

```diff
@@ -21,20 +22,58 @@ namespace Mem
 
 class PageId;
 
-/// Atomic container of "free" page numbers inside a single SharedMemory space.
-/// Assumptions: all page numbers are unique, positive, have an known maximum,
-/// and can be temporary unavailable as long as they are never trully lost.
+/// reflects the dual nature of PageStack storage:
+/// - for free pages, this is a pointer to the next free page
+/// - for used pages, this is a "used page" marker
+class PageStackStorageSlot
+{
+public:
+    typedef uint32_t PointerOrMarker;
+    typedef PointerOrMarker Pointer;
+    typedef PointerOrMarker Marker;
+
+    /// represents a nil next slot pointer
+    static const Pointer NilPtr = std::numeric_limits<PointerOrMarker>::max();
+    /// marks a slot of a used (i.e. take()n) page
+    static const Marker TakenPage = std::numeric_limits<PointerOrMarker>::max() - 1;
+    static_assert(TakenPage != NilPtr);
```

Comment:
This does not compile with c++11. We need 2-parameterr static_assert version instead:
```suggestion
    static_assert(TakenPage != NilPtr, "unique page stack markers");
```